### ISSUE TITLE
Fullscreen exit button doesn't work in iframes / other windows

### ISSUE THREAD

**Issue by Aleafan** [igordanchenko/yet-another-react-lightbox#270]
### Describe the bug

When Lightbox is open inside iframe or other window it's not possible to exit Fullscreen mode via Fullscreen button, exit works only via Escape or browser's API special button.

### Expected behavior

Fullscreen button is fully functional in iframes / other windows.

### How to reproduce

Open Lightbox in iframe.

### Additional context

This behavior's root cause - FullscreenContextProvider adds event listeners for document when iframe's top level document is different. As a result "fullscreenchange" event is not triggered, fullscreen state is not changed and fullscreen button doesn't work as intended.

I patched it for my project by finding and using ownerDocument in some instances and it works. Here is my updated index file:
[index.txt](https://github.com/igordanchenko/yet-another-react-lightbox/files/15319687/index.txt)

I'm not sure how many people will use this in iframes or other windows, so maybe its not important for 99% of usages.

**Comment by igordanchenko** [comment]
Thank you for bringing this up. Can you share a minimal repro? I'm trying to recreate the scenario you described above, but I'm unable to reproduce the issue so far.

https://stackblitz.com/edit/yet-another-react-lightbox-270?file=src%2FApp.tsx,frame.html,index.html

**Comment by Aleafan** [comment]
I've tried to setup the repo. It seems that everything works when the script file is in iframe so it references its own document correctly. Even if I put the script out of iframe Vite for some reason still puts scripts with modules inside it and it still works.
On our project script belongs to "outer" document. 

I will get back to it tomorrow and provide more info.

Thank you!

**Comment by igordanchenko** [comment]
Thank you. Any kind of reproduction (not necessarily Vite-based) will be very helpful.

**Comment by Aleafan** [comment]
OK, I reproduced the issue. For this you have to also use Inline plugin and have script outside of iframe (I rendered React component into root inside iframe).
https://stackblitz.com/edit/yet-another-react-lightbox-270-sxd4p4?file=src%2Fmain.tsx

Another possible setup without iframes is to portal React components into other window (e.g. with https://www.npmjs.com/package/react-new-window).

Don't know if those cases justify increasing code complexity.

**Comment by igordanchenko** [comment]
Thank you. I've reduced your repro to the following case - https://stackblitz.com/edit/yet-another-react-lightbox-270-1?file=src%2FApp.tsx

Just out of curiosity, what is the use case for this kind of setup?

**Comment by Aleafan** [comment]
We use this setup for encapsulating rendered trees with React elements from another context.

We also have a feature of opening pieces of UI in separate window where I first noticed this problem.

**Comment by igordanchenko** [comment]
These use cases definitely add a wrinkle to the whole implementation, as any reference to the `window` or `document` globals may be incorrect. And modular design does not make the fix any easier as some components that use `window` or `document` do not even have any DOM node reference.

I've identified some other areas exhibiting incorrect behavior due to the same underlying cause:

1) Keyboard navigation doesn't work initially when the lightbox opens in a popup window.
2) Thumbnails keyboard navigation stops working when one of the thumbnails loses focus.
3) Zoom-in via double-click / double-tap is way off when the parent window has a scroll offset.
4) The NoScroll module measures the parent window dimensions and "fixes" scrollbar padding in the parent document instead of the popup. 

And I'm not even 100% confident I've identified all relevant issues.

I'll cut a patch release once I have a solution for the above issues.
